docs/static/config-details.asciidoc

```diff
@@ -87,6 +90,35 @@ Keep the overall memory requirements in mind when you allocate memory.
 By default, a JVM's off-heap direct memory limit is the same as the heap size. Check out <<plugins-inputs-beats-memory,beats input memory usage>>.
 Consider setting `-XX:MaxDirectMemorySize` to half of the heap size.
 
+Internal JVM data structures, thread stacks, memory mapped files, direct memory for input/output (IO) operations, and the Java heap used to execute Logstash all use JVM memory. 
+Keep the memory requirements for these processes in mind as you calculate your ideal memory allocation.
+In addition to Java heap and direct memory, you must account for native memory as part of the overall memory consumption for {ls} processes.
```

Comment:
this is the first place we mention these "three kinds of memories" but we don't explain, for example, direct vs native memory. maybe the line above can be slightly altered to provide this distinction (or just make the on-heap vs off heap distinction).

Reply:
Updated by commit d692dc1466c94a2d449108b50dc9e67d1b3483a4